### ISSUE TITLE
Only fold functions at n levels deep

### ISSUE THREAD

**Issue by ProLoser** [robballou/atom-fold-functions#7]
In javascript, EVERYTHING is a function. I would like to only fold functions at 2 levels deep. This way my outer function (the class) isn't folded, but the methods are folded.

**Comment by robballou** [comment]
Yeah, this doesn't work well everywhere. I'll try to think through a good way around this.

One option that came to mind is grammar-specific settings for depth. Currently this only folds the first level that has a function... so for JavaScript this is the very top quite often (but not necessarily!). Grammar specific rules may not work because some JS may have a top level function (browser JS) and others may not (node JS).

Part of the issue is we need to scan the file to find functions which is slightly costly (scan all the rows and usually have to scan into the row to find if it has a function or not).

**Comment by ProLoser** [comment]
Use ctags?

**Comment by robballou** [comment]
Not in a release yet, but I added a `skipAutofoldWhenOnlyOneFunction`, which stops those annoying everything-is-a-function autofold! This only affects autofolding right now. I still need to figure out if there's a good way to handle specifying grammar-specific settings and for folding on those, etc when manually acting on it.

**Comment by djh1997** [comment]
is it possible to bring back folding on untabbed functions now or even a toggle between 1st and second level folding
